coalib/output/ConsoleInteraction.py

```diff
@@ -131,36 +130,61 @@ def acquire_actions_and_apply(console_printer,
     :param result:          A derivative of Result.
     :param file_dict:       A dictionary containing all files with filename as
                             key.
-    :param cli_actions:     The list of cli actions available.
     """
-    cli_actions = CLI_ACTIONS if cli_actions is None else cli_actions
     failed_actions = set()
     while True:
-        actions = []
-        for action in cli_actions:
-            if action.is_applicable(result, file_dict, file_diff_dict) is True:
-                actions.append(action)
+        actions = result.get_filtered_actions(file_dict, file_diff_dict)
 
         if actions == []:
             return
 
         action_dict = {}
         metadata_list = []
+        # FIXME: If we want to allow users to add multiple actions of same
+        # type, ``metadata.name`` should not be used as a key for action_dict
```

Comment:
Also, we'll have design a way to let the user know how are the multiple actions of same class different, because currently if we allow multiple actions, all of the actions would have the same description leading to confusion for the user.